src/rclpy/_rclpy.c.in

```diff
@@ -0,0 +1,319 @@
+#include <Python.h>
+
+#include <rmw/rmw.h>
+#include <rcl/rcl.h>
+#include <rcl/node.h>
+#include <rosidl_generator_c/message_type_support_struct.h>
+
+static PyObject *
+rclpy_init(PyObject *Py_UNUSED(self), PyObject *Py_UNUSED(args))
+{
+  /* TODO(esteve): parse args */
+  rcl_ret_t ret = rcl_init(0, NULL, rcl_get_default_allocator());
+  (void)ret;
+  Py_RETURN_NONE;
+}
+
+static PyObject *
+rclpy_create_node(PyObject *Py_UNUSED(self), PyObject *args)
+{
+  const char *node_name;
+
+  if (!PyArg_ParseTuple(args, "s", &node_name))
+  {
+    return NULL;
+  }
+
+  rcl_node_t *node = PyMem_Malloc(sizeof(rcl_node_t));
+  node->impl = NULL;
+  rcl_node_options_t default_options = rcl_node_get_default_options();
+  rcl_ret_t ret = rcl_node_init(node, node_name, &default_options);
+  (void)ret;
+  PyObject *pynode = PyCapsule_New(node, NULL, NULL);
+  return pynode;
+}
+
+static PyObject *
+rclpy_create_publisher(PyObject *Py_UNUSED(self), PyObject *args)
+{
+  PyObject *pynode;
+  PyObject *pymsg_type;
+  PyObject *pytopic;
+
+  if (!PyArg_ParseTuple(args, "OOO", &pynode, &pymsg_type, &pytopic))
+  {
+    return NULL;
+  }
+
+  assert(PyUnicode_Check(pytopic));
+
+  char *topic = (char *)PyUnicode_1BYTE_DATA(pytopic);
+
+  rcl_node_t *node = PyCapsule_GetPointer(pynode, NULL);
+
+  PyObject *pymetaclass = PyObject_GetAttrString(pymsg_type, "__class__");
+
+  PyObject *pyts = PyObject_GetAttrString(pymetaclass, "_TYPE_SUPPORT");
+
+  rosidl_message_type_support_t * ts = PyCapsule_GetPointer(pyts, NULL);
+
+  rcl_publisher_t *publisher = PyMem_Malloc(sizeof(rcl_publisher_t));
+  publisher->impl = NULL;
+  rcl_publisher_options_t publisher_ops = rcl_publisher_get_default_options();
+  rcl_ret_t ret = rcl_publisher_init(publisher, node, ts, topic, &publisher_ops);
+  (void)ret;
```

Comment:
The function warns for not checking the return value for a reason. Instead of silencing the warning here I think it should either not catch the return value (and result in a compiler warning) or this needs a TODO.